### ISSUE TITLE
Cancel or kill a long-running query?

### ISSUE THREAD

**Issue by kkuchta** [psequel/psequel-issues#39]
Forgive me if this already exists, but I can't seem to find any way to kill a long-running query.  If I accidentally run a query that'll take 20 minutes to complete, is there any way to kill it before it finishes?

If not, I would strongly request such a feature.  Messing with queries to improve their performance is one of the main use cases I have for any SQL GUI tool, and that often results in long-running queries.

Besides that, though, this is a great tool and I really appreciate that you've built such a great thing.  Thanks!

**Comment by ofirbb** [comment]
+1

**Comment by bwheeler96** [comment]
+1

**Comment by jayshepherd** [comment]
+1

**Comment by kowalcj0** [comment]
+1

**Comment by psequel** [comment]
V1.2.7 add a preference for setting query timeout. If a query takes more than the specified number of seconds to complete, it will get cancelled automatically. In future, I'll add UI for canceling a query manually.

**Comment by kkuchta** [comment]
Awesome!  The UI for canceling would be preferable, but the preference goes a long way towards solving this use case for me.  Thanks!

**Comment by rdp** [comment]
nice thanks! (might be still nice to add them to the dropdown query history even if they timeout, but that's a separate issue I suppose).

**Comment by pate** [comment]
@psequel please add this. It seems like such a core feature to dealing with any kind of database queries. A running  query blocks the whole UI. So while I'm waiting for however many seconds, I can't amend my query for the next running or copy the text.

**Comment by rdp** [comment]
Yes a cancel button on the "running query" dialog would be nice.  In my perfect world it wouldn't (by default) timeout queries, but instead just prompt, after a timeout "this is taking forever, cancel?" for the case of queries that genuinely need to use more than the setting, FWIW.

**Comment by vinibrsl** [comment]
Are we going to have this? :shipit:

**Comment by mdorda** [comment]
Any update?
